### PR TITLE
CLIENT-6779 | Fix emitting of audio level warnings

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [CLIENT-6779](https://issues.corp.twilio.com/browse/CLIENT-6779)

### Description

Fix an issue where constant audio input level warnings are not emitted.
Rename RTC Monitor to StatsMonitor, since we are now including audio levels in the stats.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Squashed erroneous commits if necessary
* [ ] Re-tested if necessary

### CODE REVIEW

File: lib/twilio/statsMonitor.ts [hunk 1]
```diff
@@ -215,6 +228,16 @@ class RTCMonitor extends EventEmitter {
     return !!this._activeWarnings.get(warningId);
   }
 
+  /**
+   * Called when a volume sample is available
+   * @param inputVolume - Input volume level from 0 to 32767
+   * @param outputVolume - Output volume level from 0 to 32767
+   */
+  onVolume(inputVolume: number, outputVolume: number): void {
```
Comment (ryan-rowland) [inline lib/twilio/statsMonitor.ts]: We should avoid the `on*` naming scheme here. This is generally used to set event handler callbacks (an alternative to using event emitter).

How about `addVolumes`?

File: lib/twilio/statsMonitor.ts [hunk 2]
```diff
@@ -260,6 +283,11 @@ class RTCMonitor extends EventEmitter {
    * @returns A universally-formatted version of RTC stats.
    */
   private _createSample(stats: IRTCStats, previousSample: RTCSample | null): RTCSample {
+    const audioInputLevel = Math.round(average(this._inputVolumes));
+    const audioOutputLevel = Math.round(average(this._outputVolumes));
+    this._inputVolumes.splice(0);
+    this._outputVolumes.splice(0);
```
Comment (ryan-rowland) [inline lib/twilio/statsMonitor.ts]: We could probably concatenate these 4 lines into 2 right?

Comment (ryan-rowland) [reply lib/twilio/statsMonitor.ts]: What value is set if the arrays are empty? What's the output of `Math.round(average([]))`?

Comment (charliesantos) [reply lib/twilio/statsMonitor.ts]: values will be zero.

**Pull request merged**